examples/rust/viewer_callbacks/src/main.rs

```diff
@@ -39,31 +39,26 @@ async fn main() -> Result<(), Box<dyn std::error::Error>> {
     let shared_state: Arc<Mutex<SharedState>> = Default::default();
 
     let startup_options = re_viewer::StartupOptions {
-        callbacks: Some(
-            re_viewer::Callbacks::builder()
-                .on_selection_change({
-                    let shared_state = shared_state.clone();
-                    move |items| {
-                        shared_state.lock().current_selection = items;
-                    }
-                })
-                .on_timeline_change({
-                    let shared_state = shared_state.clone();
-                    move |timeline, time| {
-                        let mut shared_state = shared_state.lock();
-                        shared_state.current_timeline = timeline.name().as_str().to_owned();
-                        shared_state.current_time = time.as_f64();
-                    }
-                })
-                .on_time_update({
-                    let shared_state = shared_state.clone();
-                    move |time| {
-                        let mut shared_state = shared_state.lock();
-                        shared_state.current_time = time.as_f64();
-                    }
-                })
-                .build(),
-        ),
+        on_event: Some({
+            let shared_state = shared_state.clone();
+            Arc::new(move |event: ViewerEvent| match event.kind {
+                ViewerEventKind::Play | ViewerEventKind::Pause => {}
+                ViewerEventKind::TimeUpdate { time } => {
+                    shared_state.lock().current_time = time.as_f64();
+                }
+                ViewerEventKind::TimelineChange {
+                    timeline_name,
+                    time,
+                } => {
+                    let mut shared_state = shared_state.lock();
```

Comment:
nit: this could be moved to before the `match` so that we don't have to lock in each branch